CONTRIBUTING.md

```diff
@@ -58,3 +58,25 @@ import Foo as F
 import Qux (Q(..), q)
 import qualified A
 ```
+
+## Development Workflow
+
+Here we outline a general workflow new contributors could adopt for a nice incremental development experience.
+
+The root folder consists of two stack-*.yaml config files.
+
+1. stack-night.yaml - Config file defines all the settings to build and test any changes.
```

Comment:
Side note: Maybe we should rename stack-nightly.yaml to just stack.yaml so you don't need to pass `--stack-yaml`. I think the reason we changed from the default name was that we had several more of these files previously, but it was cumbersome to maintain all of them.

Reply:
I'm slightly in favor of *not* having a `stack.yaml` because that allows you to have a custom `stack.yaml` without introducing edits tracked by git. In particular, by reusing the same resolver, dependencies can be shared across multiple projects. If that's not a problem, it seems simple enough to ask users to make a symbolic link or a copy.

Reply:
works for me!